code/modules/supermatter/supermatter.dm

```diff
@@ -462,4 +463,102 @@
 /obj/machinery/power/supermatter/clone(var/obj/machinery/power/supermatter/O)
 	id_tag = O.id_tag
 	set_frequency(O.frequency)
-	return 1
\ No newline at end of file
+	return 1
+
+/obj/machinery/computer/supermatter
+	name = "supermatter monitoring computer"
+	desc = "Monitors ambient temperature and stability of a linked shard to provide early warning information regarding delamination. Can be linked up to supermatter with a matching frequency and id_tag using a multitool. While using a multitool on supermatter is safe, Nanotrasen accepts no responsibility or liability for sudden rushes of wind or radiation poisoning."
+	icon_state = "sme"
+	circuit = "/obj/item/weapon/circuitboard/supermatter"
+	light_color = LIGHT_COLOR_YELLOW
+	var/frequency = 1333
+	var/id_tag //mappable
+	var/datum/radio_frequency/radio_connection
+	var/obj/machinery/power/supermatter/linked = null
```

Comment:
I believe the shard you're keeping a reference to here won't be able to be GC'd. You could use the `on_destroyed` event of the shard to clear this reference.

Reply:
Gotcha, I'll get to all this tonight